frontend/src/components/Legend/Legend.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ReactElement, useState } from 'react';
+
+interface DescriptionProps {
+  name: string;
+  description: string;
+  imgURL: string;
+}
+
+const Description = (props: DescriptionProps) => {
+  const { name, description, imgURL } = props;
+  return (
+    <div className="grid grid-cols-4 text-center justify-items-center p-2">
+      <div className="col-span-3">
+        <p className="font-bold">{name}</p>
+        <p className="text-l">{description}</p>
+      </div>
+      <img height={60} width={60} src={imgURL} alt={name}></img>
+    </div>
+  );
+};
+
+const Legend: React.FunctionComponent = (): ReactElement => {
+  const [isLegendToggled, setIsLegendToggled] = useState(false);
+
+  const toggleLegend = () => {
+    setIsLegendToggled(true);
+  };
+
+  const untoggleLegend = () => {
+    setIsLegendToggled(false);
+  };
```

Comment:
it can be achieved in a single function eg. handleLegendToggle - set state to !isLegendToggled